File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        Stack<Character> stack = new Stack<>();

        for (char c: s) {
            stack.push(c);
        }

        int i = 0;
        while (!stack.isEmpty()) {
            s[i] = stack.pop();
            i++;
        }
    }
}

File: 0344-reverse-string/0344-reverse-string.kt
class Solution {
    fun reverseString(s: CharArray): Unit {
        var left = 0
        var right = s.size - 1
        
        while (left < right) {
            s[left] = s[right].also { s[right] = s[left] }
            right--
            left++
        }
    }
}